Assets/_Laz/Tests/CheckpointManagerTests.cs

```diff
@@ -0,0 +1,85 @@
+    {
+
+        private LazPlayer _player = new LazPlayer();
+        private CheckPointManager _checkPointManager = null;
+        private LazCoordinatorBehaviour _lazCoordinatorBehaviour = null;
+        private MockLazMovement _mockMovement = new MockLazMovement();
+        private MockLazoProperties _lazoProperties = new MockLazoProperties();
+        private Keyboard _keyboard = null;
+
+        [SetUp]
+        public override void Setup()
+        {
+            base.Setup();
+            _mockMovement = new MockLazMovement();
+            _lazoProperties = new MockLazoProperties();
+            _player = new LazPlayer();
+            EditorSceneManager.LoadSceneAsyncInPlayMode("Assets/_Laz/Scenes/TestingScenes/LazCheckPointManagerTests.unity", new LoadSceneParameters(LoadSceneMode.Single));
+            _keyboard = InputSystem.AddDevice<Keyboard>();
+        }
+
+        [UnityTest]
+        public IEnumerator Test_CheckPointManager_TestInitialize()
+        {
+            for (int i = 0; i < 5; i++)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+
+            // When
+            _checkPointManager = GameObject.FindObjectOfType<CheckPointManager>();
+
+            // Then
+            yield return new WaitForSeconds(0);
+            _checkPointManager.Initialize(_player);
+
+            // Therefore
+            Vector3 _initialPosition = new Vector3(-10f, 0f, 0f);
+            var actualSpawnPosition = _player.SpawnPosition;
+            Assert.AreEqual(_initialPosition, actualSpawnPosition, "Spawn position is not set to initial checkpoint position!");
+        }
+
+        [UnityTest]
+        public IEnumerator Test_CheckPointManager_TestSetNewCheckpoint()
+        {
+            for (int i = 0; i < 5; i++)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+
+            // When
+            _lazCoordinatorBehaviour = GameObject.FindObjectOfType<LazCoordinatorBehaviour>();
+            _checkPointManager = GameObject.FindObjectOfType<CheckPointManager>();
+            _lazCoordinatorBehaviour.gameObject.transform.position = Vector3.zero;
+
+            _lazoProperties.DistanceLimitOfLazo = float.MaxValue;
+            _lazoProperties.TimeToLivePerPoint = float.MaxValue;
+            _lazCoordinatorBehaviour.Initialize(_player, new ILazoWrapped[] { }, _mockMovement, _lazoProperties);
+            _checkPointManager.Initialize(_player);
+
+            // Then
+            Press(_keyboard.dKey);
+            yield return new WaitForSeconds(2.5f);
+            Release(_keyboard.dKey);
+            yield return new WaitForSeconds(0);
+
+            // Therefore
+            Vector3 _initialPosition = new Vector3(10f, 0f, 0f);
+            var actualSpawnPosition = _player.SpawnPosition;
+            Assert.AreEqual(_initialPosition, actualSpawnPosition, "Spawn position is not set to new checkpoint position!");
+        }
+    }
+}
```

Comment:
i would add one more test, where it goes to the right checkpoint, then back to the LHS checkpoint,
to prove that you could reactivate an older checkpoint

Reply:
Ah good idea, I'll add that